src/wasm-type.h

```diff
@@ -64,16 +72,53 @@ class Type {
   const std::vector<Type>& expand() const;
 
   // Predicates
+  //                     C       C
+  //                     o       o
+  //                     m   S   n
+  //                 B   p   i   c
+  //                 a   o   n   r
+  //                 s   u   g   e
+  //                 i   n   l   t
+  //   Type          c   d   e   e
+  // ╞═════════════╬═══╪═══╪═══╪═══╪═══════╡
```

Comment:
I think this would be easier to read.

```suggestion
  //                 
  //                 ┌ Basic
  //                 |   ┌ Compound
  //                 |   |   ┌ Single
  //   Type          |   |   |   ┌ Concrete  
  // ╞═════════════╬═══╪═══╪═══╪═══╪═══════╡
```